src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java

```diff
@@ -593,6 +604,36 @@ public Collection<Flow> findLoopedByLoopSwitchId(SwitchId switchId) {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public Map<Integer, Collection<Flow>> findSwitchFlowsByPort(SwitchId switchId, Collection<Integer> ports) {
+        try {
+            @SuppressWarnings("unchecked")
+            Map<Integer, Collection<Flow>> portToFlowMap = collectToPortToFlowTreeMap(
+                    framedGraph().traverse(g -> g.V()
+                            .hasLabel(PathSegmentFrame.FRAME_LABEL)
+                            .or(
+                                    has(PathSegmentFrame.SRC_SWITCH_ID_PROPERTY, switchId.toString()),
+                                    has(PathSegmentFrame.DST_SWITCH_ID_PROPERTY, switchId.toString()))
+                            .as(PATH_SEGMENT_ALIAS)
+                            .in(FlowFrame.OWNS_SEGMENTS_EDGE)
+                            .in(FlowFrame.OWNS_SEGMENTS_EDGE)
```

Comment:
Could you pleaseu use "owns" from FlowFrame?

Reply:
fixed. I also removed the constant from FlowFrame